### PR TITLE
LMER example

### PR DESCRIPTION
**Pull request by ae-foster**
Adds a hierarchical logistic regression model.

### CODE REVIEW

File: examples/contrib/oed/item_response.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    """Hierarchical logistic regression model.
+
+    :param torch.tensor design_tensor: a `batch_dims x n x p`
+        tensor giving the features of the `n` responses (e.g.
+        the features of the `n` objects to be given at this time)
+    :param list participant_id: a hierarchical list of the groups
+        that the current participant is assigned to. The first element
+        indexes the highest level group. For instance, a geographical
+        hierarchy list could look like 
+        `["United States", "California", "Santa Barbara"]`
```
Comment (martinjankowiak) [inline examples/contrib/oed/item_response.py]: santa barbara is about 200 miles too far to the south to be welcome in this codebase ; )

File: examples/contrib/oed/item_response.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+# Number of layers of hierarchy (1 layer- classical linear model)
+layers = 2
+# Parameters controlling the prior variance
+A = 20.
+B = 10.
+beta = torch.tensor(1.)
+
+softplus = torch.nn.functional.softplus
+sigmoid = torch.nn.functional.sigmoid
```
Comment (eb8680) [inline examples/contrib/oed/item_response.py]: nit: `from torch.nn.functional import softplus, sigmoid`

File: examples/contrib/oed/item_response.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        design_tensor = spherical_design_tensor(designs)
+        print('design tensor', design_tensor.shape)
+        est_ape = vi_ape(
+            lambda d: model(d, participant),
+            design_tensor,
+            observation_labels="y",
+            vi_parameters={
+                "guide": lambda d: guide(d, participant),
+                "optim": optim.Adam({"lr": 0.0025}),
+                "loss": Trace_ELBO(),
```
Comment (eb8680) [inline examples/contrib/oed/item_response.py]: Prefer `TraceEnum_ELBO(strict_enumeration_warning=False).differentiable_loss`

File: examples/contrib/oed/item_response.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        intercept += pyro.sample(layer_name + "_intercept", intercept_dist)
+
+        coef_sd = torch.ones(coef_shape)*B*torch.exp(-beta*layer)
+        coef_dist = dist.Normal(0., coef_sd).independent(2)
+        coef += pyro.sample(layer_name + "_coef", coef_dist)
+
+    logit_p = torch.matmul(design_tensor, coef.t()).squeeze(-1) + intercept
+    p = sigmoid(logit_p)
+    # Binary outcomes - responses to `n` items shown to given participant
+    pyro.sample("y", dist.Bernoulli(p).independent(1))
```
Comment (eb8680) [inline examples/contrib/oed/item_response.py]: `return pyro.sample(...`

File: examples/contrib/oed/item_response.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+
+def guide(design_tensor, participant_id):
```
Comment (eb8680) [inline examples/contrib/oed/item_response.py]: Let's try using an autoguide instead

File: examples/contrib/oed/item_response.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                "loss": Trace_ELBO(),
+                "num_steps": num_vi_steps},
+            is_parameters={"num_samples": 10},
+            y_dist=ydist
+        )
+        return est_ape
+
+    participant = ["a"]
+    X = torch.tensor([[0., 0.], [0., 1.5]])
+    y = estimated_ape(X, participant, None)
```
Comment (eb8680) [inline examples/contrib/oed/item_response.py]: We should try using a real dataset here instead

File: examples/contrib/oed/item_response.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        intercept_sd = torch.ones(response_shape)*A*torch.exp(-beta*layer)
+        intercept_dist = dist.Normal(0., intercept_sd).independent(1)
+        intercept += pyro.sample(layer_name + "_intercept", intercept_dist)
+
+        coef_sd = torch.ones(coef_shape)*B*torch.exp(-beta*layer)
+        coef_dist = dist.Normal(0., coef_sd).independent(2)
+        coef += pyro.sample(layer_name + "_coef", coef_dist)
+
+    logit_p = torch.matmul(design_tensor, coef.t()).squeeze(-1) + intercept
+    p = sigmoid(logit_p)
```
Comment (eb8680) [inline examples/contrib/oed/item_response.py]: You can just pass `logit_p` directly with `Bernoulli(logits=logit_p)`
